### PR TITLE
NYC resident tax

### LINKED ISSUE #1282: NYC resident tax
https://www.tax.ny.gov/pit/file/tax_tables.htm

https://codelibrary.amlegal.com/codes/newyorkcity/latest/NYCadmin/0-0-0-13462

### PR DESCRIPTION
**Pull request by Aleida24**
Fixes #1282

Thanks for contributing! Please remove any top-level sections that do not apply to your changes.

- [ ] `make format && make documentation` has been run.

# New variable

- [ ] Label field added
- [ ] Documentation field added
- [ ] Unit field added
- [ ] Default value field added if relevant
- [ ] Variable name follows conventions
- [ ] Unit test(s) added
- [ ] Integration test(s) added if relevant
- [ ] Issues this PR fixes linked

## What's changed

Description of the changes here.

# Bug fix

- [ ] Regression test added
- [ ] Regression test passing

## What this fixes and how it's fixed

Description of how this fix works goes here. Link any issues this PR fixes.

### CODE REVIEW

**Review by Aleida24 — COMMENTED** [review]
added NYC resident taxes

**Comment by MaxGhenis** [comment]
I think we're hitting test failures because `county` is wrapped as an array rather than an openfisca [`enum_array`](https://github.com/openfisca/openfisca-core/blob/443bb44c6ae1a48263ee4c25faac7af01f99632d/openfisca_core/indexed_enums/enum_array.py).

@Aleida24 could you find legislative references for the parameters? I'm especially interested in whether the NYC tax rates are specified in the NYS or NYC legal code.

File: openfisca_us/variables/gov/local/ny/nyc/tax/income/ny_nyc_income_tax_before_credits.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    entity = TaxUnit
+    label = "NYC income tax before credits"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.NY
+
+    def formula(tax_unit, period, parameters):
+        # determine if tax unit is in NYC
+        county = tax_unit.household("county", period)
+        counties = county.possible_values
```
Comment (nikhilwoodruff) [inline openfisca_us/variables/gov/local/ny/nyc/tax/income/ny_nyc_income_tax_before_credits.py:15]: To take a look at

Comment (MaxGhenis) [reply openfisca_us/variables/gov/local/ny/nyc/tax/income/ny_nyc_income_tax_before_credits.py:15]: This is hitting #1342 too @PavelMakarchuk

**Comment by MaxGhenis** [comment]
Sorry this took so long @Aleida24, thank you for this contribution! I updated it for the new framework and to use `in_nyc` to get around the county issue (see #1734). Merging!

**Pull request merged**